[62823] NoMethodError in WorkPackages::DatePickerController#show

HDinger opened the pull request:
# Ticket
https://community.openproject.org/projects/openproject/work_packages/62823/activity

# What are you trying to accomplish?
Sometimes the daepicker_controler throws a `undefined method '[]' for nil` error. Since we did not manage to reproduce the issue itself, this is a cheap "fix" by preventing that `nil` case

HDinger committed:
Catch nil case

cbliard reviewed (changes requested):
I tried hard to reproduce this one, but did not manage so far...

I think your fix is valid, but we could move the logic up.
It all happens because wp_params is `nil`. It's computed in `work_package_datepicker_params`. It's `nil` because it's
```
def work_package_datepicker_params
  if params[:work_package]
    params.require(:work_package).etc.etc
  end
end
```

I think adding a `else` clause returning `{}` like [you did](https://github.com/opf/openproject/commit/fdc66f856b7360223d762a7fdc8f624f3fd52ebf#diff-1e9db212c812d112e328725bc3aba1171f49045df793e9bd108e5a8e0b8f8c0cR127-R128) in `#touched_field_map` would be a more durable fix. (By any chance, do you remember what was the buggy state that made you add this `else` clause back then?).

Could you do that please?

HDinger commented:
> I tried hard to reproduce this one, but did not manage so far...
> 
> I think your fix is valid, but we could move the logic up. It all happens because wp_params is `nil`. It's computed in `work_package_datepicker_params`. It's `nil` because it's
> 
> ```
> def work_package_datepicker_params
>   if params[:work_package]
>     params.require(:work_package).etc.etc
>   end
> end
> ```
> 
> I think adding a `else` clause returning `{}` like [you did](https://github.com/opf/openproject/commit/fdc66f856b7360223d762a7fdc8f624f3fd52ebf#diff-1e9db212c812d112e328725bc3aba1171f49045df793e9bd108e5a8e0b8f8c0cR127-R128) in `#touched_field_map` would be a more durable fix. (By any chance, do you remember what was the buggy state that made you add this `else` clause back then?).
> 
> Could you do that please?

Done 👍 

unfortunately, I can't remember why I added that back then.. I added that shortly after starting on the datepicker, so many things have changed since then. It could very well be that the error I had back then, does not exist any more 🤷‍♀️

cbliard reviewed (approved):
I could "reproduce" it by manually going to http://localhost:3000/work_packages/46/datepicker_dialog_content/edit?triggering_field=due_date
It does not trigger any error anymore, so I consider it fixed.

Nice job 👍

cbliard commented:
The failing specs are flickering and are not related to this PR anyway. 
Merging.
